src/beignet/_apply_transform.py

```diff
@@ -0,0 +1,114 @@
+import torch
+from torch import Tensor
+from torch.autograd import Function
+
+
+def _apply_transform(input: Tensor, transform: Tensor) -> Tensor:
+    """
+    Applies an affine transformation to the position vector.
+
+    Parameters
+    ----------
+    input : Tensor
+        Position, must have the shape `(..., dimension)`.
+
+    transform : Tensor
+        The affine transformation matrix, must be a scalar, a vector, or a
+        matrix with the shape `(dimension, dimension)`.
+
+    Returns
+    -------
+    Tensor
+        Affine transformed position vector, has the same shape as the
+        position vector.
+    """
+    if transform.ndim == 0:
+        return input * transform
+
+    indices = [chr(ord("a") + index) for index in range(input.ndim - 1)]
+
+    indices = "".join(indices)
+
+    if transform.ndim == 1:
+        return torch.einsum(
+            f"i,{indices}i->{indices}i",
```

Comment:
I think we can use ellipsis (...) here instead of explicitly constructing the indices

Reply:
Nice